protocols/purple/purple.c

```diff
@@ -331,6 +346,12 @@ static int purple_buddy_msg(struct im_connection *ic, char *who, char *message,
 	PurpleConversation *conv;
 	struct purple_data *pd = ic->proto_data;
 
+	if (!strncmp(who, PURPLE_REQUEST_HANDLE, strlen(PURPLE_REQUEST_HANDLE))) {
+		guint request_id = atoi(who + strlen(PURPLE_REQUEST_HANDLE) + 1);
```

Comment:
Why calculate the length of a static string repeatedly?  Just create another macro or a local static variable.